docs/mutations.md

```diff
@@ -269,13 +348,127 @@ print(f"Before: {mts.num_mutations} mutations.\n"
       f"After: {mmts.num_mutations} mutations.")
 ```
 
-Adding ancestral mutations can result in confusing situations, so should be
-done with care. For instance, new mutations added to an already mutated site
+This might be done, for instance, if a tree sequence is the result of a forwards
+simulation that has non-neutral variation, and you'd like to add neutral mutations.
+
+(sec_mutations_state_independence)=
+
+### Silent mutations and state-independence
+
+However, adding mutations in a way that might result in a new mutation being
+*ancestral* to an existing one can result in confusing situations.
+For instance, new mutations added to an already mutated site
 will not assign a new ancestral state. Furthermore, if mutations are added to
-a tree sequence under two different models, impossible transitions may result.
-See {func}`.sim_mutations` for further explanation. However, typical usage
-with low per-base pair mutation rates and nucleotide models will result
-in very few of these cases.
+a tree sequence already having some, unlikely transitions may result.
+For instance, in the {class}`.PAMMutationModel`,
+codon mutations from ``V`` to ``G`` and ``V`` to ``I`` are reasonably likely,
+but a mutation from ``G`` to ``I`` is impossible.
+So, applying this model to the same tree sequence twice could insert a
+V->G mutation above a pre-existing V->I mutation,
+resulting in the chain V->G->I, which is impossible under this model
+(the chain is possible under the {class}`.BLOSUM62MutationModel`, but unlikely).
+In practice this is more confusing than concerning,
+since it can only happen when the same site is mutated more than once,
+which affects only a small proportion of sites.
+
+The situation above occurred because the probability of mutating to ``I``
+was very different depending on the initial state.
+However, if derived alleles are chosen in a way *independent* of the previous state,
+then adding mutations at rate :math:`\\alpha`
+and then adding mutations at rate :math:`\\beta`
+is *equivalent* to adding mutations once, at rate :math:`\\alpha + \\beta`.
+A mutation model has this property if rows of its transition matrix are the same,
+and is called *state independent* (or, *parent independent*).
+For instance, mutations under the {class}`.JC69MutationModel`
+choose *any* derived allele with equal probability.
+This means that the *result* of a mutation can be chosen without knowing
+what the previous state at that site was ("state independence"),
+which implies that adding new mutations above a given mutation does not affect
+the probability of that mutation ocurring.
+However, this is only true if "any derived allele" includes the previous allele,
+i.e., if we include *silent mutations*.
```

Comment:
good to link back to the silent mutations section, {ref}`silent mutations<sec_mutations_silent_mutations>`. As I say, we the sections to be fairly independent, so linking out is good.